aten/src/ATen/native/SummaryOps.cpp

```diff
@@ -52,10 +52,79 @@ Tensor _bincount_cpu_template(
   }
   return output;
 }
+
+// A helper function to get the bin for histogram
+// while each bin is inclusive at the lower end and exclusive at the higher,
+// i.e. [start, end) the last bin is inclusive at both, i.e. [start, end], in
+// order to include maxvalue if exists
+template <typename input_t>
+inline int64_t getbin(input_t x, input_t min, input_t max, int64_t bins) {
+  if (x == max)
```

Comment:
Should this be `(x >= max)` for safety? Testing equality of float types always worries me.

Reply:
Alternatively this check can be moved so the following computation is clamped to the number of bins - 1.